rules/prefer-includes.js

```diff
@@ -18,7 +18,11 @@ const getProblem = (context, node, target, argumentsNodes) => {
 	const tokenStore = sourceCode.parserServices.getTemplateBodyTokenStore?.() ?? sourceCode;
 
 	const memberExpressionNode = target.parent;
-	const dotToken = tokenStore.getTokenBefore(memberExpressionNode.property);
+	let dotToken = tokenStore.getTokenBefore(memberExpressionNode.property);
+	if (dotToken.value !== '.') {
+		dotToken = sourceCode.getTokenBefore(memberExpressionNode.property);
+	}
```

Comment:
Hello @ota-meshi, sorry for bothering you.

In this case I need find the dot in

```
foo.indexOf
```

When it's in element attribute eg `v-if`, `sourceCode.parserServices.getTemplateBodyTokenStore().getTokenBefore(memberExpressionNode.property)`, I get the correct token, but when it's in `<script>`, I get the wrong token, what's the correct approach here? Do I need check if the JS code in attribute?

Reply:
Thanks, so we are using it all wrong, I'll update.